allennlp/modules/transformer/self_attention.py

```diff
@@ -13,14 +13,27 @@ class SelfAttention(TransformerModule, FromParams):
     Details in the paper:
     [BERT: Pre-training of Deep Bidirectional Transformers for Language Understanding, Devlin et al, 2019]
     (https://api.semanticscholar.org/CorpusID:52967399)
+
+    # Parameters
+
+    hidden_size: `int`
+    num_attention_heads: `int`
+    dropout: `float` (default = `0.0`)
+    scoring_func: `str` (default = `scaled_dot_product`)
+        The name of the attention-calculating function to be used.
+        Eg. `additive`, `linear`, etc. For a complete list, please check :mod:`allennlp.modules.attention`.
     """
 
+    _relevant_module = ["encoder.layers.0.attention.self", "encoder.layers.0.attention"]
```

Comment:
It occurs to me that if we pass in a BART model, which has a separate encoder and decoder, then this mapping wouldn't work. Am I wrong about that?

Reply:
Yes. If the `_relevant_module` isn't found, the code will return the module that was passed in, which will eventually fail. The goal is to ensure that we don't require too many one-off specifications for different variants, but some will be inevitable.